2_process/src/process_data.R

```diff
@@ -0,0 +1,43 @@
+
+process_data <- function(){
+  
+  save_path <- "./2_process/out"
+  
+  # Prepare the data for plotting:
+  eval_data <- readr::read_csv(file.path("./1_fetch/out", 'model_RMSEs.csv'), col_types = 'iccd') %>%
+    filter(str_detect(exper_id, 'similar_[0-9]+')) %>%
+    mutate(col = case_when(
+      model_type == 'pb' ~ '#1b9e77',
+      model_type == 'dl' ~'#d95f02',
+      model_type == 'pgdl' ~ '#7570b3'
+    ), pch = case_when(
+      model_type == 'pb' ~ 21,
+      model_type == 'dl' ~ 22,
+      model_type == 'pgdl' ~ 23
+    ), n_prof = as.numeric(str_extract(exper_id, '[0-9]+')))
+  
+  # Save the processed data:
+  readr::write_csv(eval_data, file = file.path(save_path, 'model_summary_results.csv'))
+  
+  # Save the model diagnostics:
+  render_data <- list(pgdl_980mean = filter(eval_data, model_type == 'pgdl', exper_id == "similar_980") %>% pull(rmse) %>% mean %>% round(2),
+                      dl_980mean = filter(eval_data, model_type == 'dl', exper_id == "similar_980") %>% pull(rmse) %>% mean %>% round(2),
+                      pb_980mean = filter(eval_data, model_type == 'pb', exper_id == "similar_980") %>% pull(rmse) %>% mean %>% round(2),
+                      dl_500mean = filter(eval_data, model_type == 'dl', exper_id == "similar_500") %>% pull(rmse) %>% mean %>% round(2),
+                      pb_500mean = filter(eval_data, model_type == 'pb', exper_id == "similar_500") %>% pull(rmse) %>% mean %>% round(2),
+                      dl_100mean = filter(eval_data, model_type == 'dl', exper_id == "similar_100") %>% pull(rmse) %>% mean %>% round(2),
+                      pb_100mean = filter(eval_data, model_type == 'pb', exper_id == "similar_100") %>% pull(rmse) %>% mean %>% round(2),
+                      pgdl_2mean = filter(eval_data, model_type == 'pgdl', exper_id == "similar_2") %>% pull(rmse) %>% mean %>% round(2),
+                      pb_2mean = filter(eval_data, model_type == 'pb', exper_id == "similar_2") %>% pull(rmse) %>% mean %>% round(2))
```

Comment:
You should make a function for every repeated expression. So here we see the same filter command applied to each of the datasets. The filter command should be a function. That way if we ever decide to change the filter, we just change the function once

Reply:
OK, good idea! The "2_process" phase code now contains a filter_data function that does the same command. For this project it seems like we know the specific model types and experiment id's we're after, but I have some ideas for how we could summarize model performance without hard-coding in the model names. We could revisit down the line if useful.